pages/index.js

```diff
@@ -1,14 +1,51 @@
 import RemediesList from "@/components/RemediesList";
 import RemedyForm from "@/components/RemedyForm";
+import FilterList from "@/components/FilterRemedies";
+import { useState } from "react";
+
 export default function HomePage({ remedies, handleAddRemedy }) {
+  const [selectedSymptom, setSelectedSymptom] = useState("");
+  const [filteredRemedies, setFilteredRemedies] = useState(remedies);
+
+  function handleSymptomChange(event) {
+    const selected = event.target.value;
+    setSelectedSymptom(selected);
+
+    if (selected === "") {
+      setFilteredRemedies(remedies);
+    } else {
+      const filtered = remedies.filter((remedy) =>
+        remedy.symptoms.includes(selected)
+      );
+      setFilteredRemedies(filtered);
+    }
+  }
+
+  function handleClearFilter() {
+    setSelectedSymptom("");
+    setFilteredRemedies(remedies);
+  }
```

Comment:
You don't need a separate state for the `filteredRemedies`. Instead, you can derive the `filteredRemedies` based on the `selectedSymptons` and the `remedies`:

```suggestion
 const [selectedSymptom, setSelectedSymptom] = useState("");

  function handleSymptomChange(event) {
    const selected = event.target.value;
    setSelectedSymptom(selected);
  }

  function handleClearFilter() {
    setSelectedSymptom("");
  }

  const filteredRemedies = selectedSymptom
    ? remedies.filter((remedy) => remedy.symptoms.includes(selectedSymptom))
    : remedies;
```